aws/cloudformation/cloud_formation_stack.yml.erb

```
@@ -401,6 +401,45 @@ Resources:
 
 <%   if cdn_enabled && app == 'Pegasus' -%>
 <%=    component 'www_redirect', app: app -%>
+  MarketingRouterLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Description: 'Lambda for dynamically routing marketing traffic to different origins based on the URL path'
+      FunctionName: <%="#{rack_env.to_s}-MarketingRouter"%>%
+      Code: <%=zip_directory 'marketing-router'%>
+      Handler: index.handler
+      Runtime: nodejs22.x
+      Role: !GetAtt MarketingRouterLambdaRole.Arn
+      Environment:
+        Variables:
+          MARKETING_DOMAIN: dev.marketing.dev-code.org
+  MarketingRouterVersion:
+    Type: AWS::Lambda::Version
+    Properties:
+      FunctionName: !Ref MarketingRouterLambda
+  MarketingRouterLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      <%=multiple_service_role ['lambda', 'edgelambda']%>
+      Path: /
+      Policies:
+        - PolicyName: <%="#{rack_env.to_s}-MarketingLambdaExecutionPolicy"%>%
```

No need to be this verbose, as the policy is defined inside the role and isn't a standalone "managed policy" we will only ever see the policy name in the context of this role.

The role itself is going to get automatically named "<stack-name>-MarketingRouterLambdaRole-<random string>".

```suggestion
        - PolicyName: Logging
```